src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -680,6 +685,19 @@ private boolean isBlocked(InetAddress address) {
         }
 
         for (ClientBlockRegistry<InetAddress> clientBlockRegistry : ipClientBlockRegistries) {
+            List<String> ignoreHosts = ((AuthFailureListener) clientBlockRegistry).getIgnoreHosts();
+            WildcardMatcher ignoreHostMatcher = WildcardMatcher.NONE;
+            if (ignoreHosts != null && !ignoreHosts.isEmpty()) {
+                ignoreHostMatcher = WildcardMatcher.from(ignoreHosts);
+            }
+            if (address != null) {
+                final String ipAddress = address.getHostAddress();
+                final String hostname = address.getHostName();
+
+                if (ignoreHostMatcher.test(ipAddress) || ignoreHostMatcher.test(hostname)) {
+                    return false;
+                }
+            }
```

Comment:
nit: should this be pulled out in a method?

Reply:
Yes, pulled this out and added tests specifically for this logic.